Signal/src/environment/NotificationsManager.m

```diff
@@ -343,8 +345,8 @@ - (void)notifyUserForIncomingMessage:(TSIncomingMessage *)message
         } else {
             if (shouldPlaySound && [Environment.preferences soundInForeground]) {
                 OWSSound sound = [OWSSounds notificationSoundForThread:thread];
-                // We play the "quiet" variation of sounds if possible for notifications in the foreground.
-                [OWSSounds playSound:sound quiet:YES shouldRespectSilentSwitch:YES];
```

Comment:
This class is the only place we were using the shouldRespectSilentSwitch behavior, we were never setting it to NO. 

So instead I wrapped it behind a method and moved the OWSSounds.sharedInstance.audioPlayer to this class.